miqa/core/models/project.py

```diff
@@ -102,20 +102,17 @@ def get_status(self):
             .prefetch_related('decisions')
             .annotate(decision_count=models.Count('decisions'))
         )
-        complete_scans_in_project = (
-            scans_in_project.filter(decision_count__gt=0)
-            .annotate(
-                last_decider_id=models.Subquery(
-                    ScanDecision.objects.filter(scan__id=models.OuterRef('id'))
-                    .order_by('-created')
-                    .values('creator_id')[:1]
-                )
-            )
-            .filter(last_decider_id__in=tier_2_reviewers)
-        )
+        complete_scans_in_project = [
+            scan
+            for (scan, last_decision) in [
+                (scan, ScanDecision.objects.filter(scan=scan).order_by('created')[0])
+                for scan in scans_in_project.filter(decision_count__gt=0)
+            ]
+            if last_decision.decision == 'U' or last_decision.creator.id in tier_2_reviewers
+        ]
```

Comment:
We should do this as a single query rather than using one query per scan. I think we just need to modify the previous subquery.

Reply:
I'm having trouble writing this; I'm finding it difficult to filter on those two things with a single subquery. Could I have your help with the rewrite?

Reply:
Sure thing. Let me take a crack at it, these things take some creativity to reason out.